### PR TITLE
8341966: Broken annotated module may lead to an exception in javac

### PR DESCRIPTION
**Pull request by lahodaj**
Consider a `module-info.class`, that both:
 - is wrong, so it will produce a bad-classfile error, for example due to an illegal `requires transitive java.base;`
 - and the module is annotated

If the javac proceeds to resolve the annotations (which is not the default, but can be pushed to do it), it will crash with an `AssertionError` while trying to un-proxy the annotations. This is because there's an assert that `requestingOwner.owner.kind == MDL`, but since the module is erroneous, the kind is `ERR`, and the assert fails.

The proposal is to loosen the assert a bit, and require `requestingOwner.owner instanceof ModuleSymbol`. That should still be sufficiently strong, while permitting broken modules to be handled.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8341966](https://bugs.openjdk.org/browse/JDK-8341966): Broken annotated module may lead to an exception in javac (**Task** - P4)

### Reviewers
 * [Adam Sotona](https://openjdk.org/census#asotona) (@asotona - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/21486/head:pull/21486` \
`$ git checkout pull/21486`

Update a local copy of the PR: \
`$ git checkout pull/21486` \
`$ git pull https://git.openjdk.org/jdk.git pull/21486/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 21486`

View PR using the GUI difftool: \
`$ git pr show -t 21486`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/21486.diff">https://git.openjdk.org/jdk/pull/21486.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/21486#issuecomment-2409988802)

### CODE REVIEW

File: test/langtools/tools/javac/modules/AnnotationsOnModules.java [hunk 1]
```diff
@@ -726,6 +729,96 @@ public TestCase(String extraDecl, String decl, String use, String expectedAnnota
...
+            .outdir(libClasses)
+            .files(findJavaFiles(lib))
+            .run()
+            .writeAll();
+
+        Path modifiedModuleInfo = libClasses.resolve("module-info.class");
+        ClassModel cm1 = ClassFile.of().parse(modifiedModuleInfo);
+        byte[] newBytes = ClassFile.of().transformClass(cm1, new ClassTransform() {
+            @Override
+            public void accept(ClassBuilder builder, ClassElement element) {
```
Comment (liach) [inline test/langtools/tools/javac/modules/AnnotationsOnModules.java:755]: Is there any restriction so that you cannot use a lambda here?

Comment (lahodaj) [reply test/langtools/tools/javac/modules/AnnotationsOnModules.java]: No, fixed.

**Review by asotona — APPROVED** [review]
Looks good to me.

**Pull request closed without merging**